### PR TITLE
add python script for ensemble vortex generation

### PR DESCRIPTION
**Pull request by zacharyburnettNOAA**
draft pull request; put discussion about besttrack perturbation here

### CODE REVIEW

**Comment by WPringle** [comment]
Great, I am happy with this here. Maybe it will be easy enough just to have this as part of this repository for good. 

Do you want to keep working on this to develop different options and modularize this in a single PR?

**Comment by WPringle** [comment]
For starters I think I would like to extract "best_track_forcing" class from adcircy and make into some kind of "ATCF" or "vortex" class here.

**Comment by zacharyburnettNOAA** [comment]
> Do you want to keep working on this to develop different options and modularize this in a single PR?
> 
> For starters I think I would like to extract "best_track_forcing" class from adcircy and make into some kind of "ATCF" or "vortex" class here.

yes, that's a good idea. Lets continue working on this branch and make it a full-fledged perturber

File: ensembleperturbation/perturbation/make_storm_ensemble.py [hunk 1]
```diff
@@ -0,0 +1,765 @@
...
+    argument_parser.add_argument('end_date', nargs='?', help='end date')
+    arguments = argument_parser.parse_args()
+
+    # hardcoding variable list for now
+    variables = [
+        MaximumSustainedWindSpeed,
+        RadiusOfMaximumWinds,
+        AlongTrack,
+        CrossTrack,
+    ]
```
Comment (zacharyburnettNOAA) [inline ensembleperturbation/perturbation/make_storm_ensemble.py:755]: use class-based variable system to store all the per-variable data (for instance, `AlongTrack` contains the column name, lower and upper bounds, perturbation randomness type, and historical errors)

File: ensembleperturbation/perturbation/make_storm_ensemble.py [hunk 2]
```diff
@@ -0,0 +1,765 @@
...
+    LINEAR = 'linear'
+
+
+class BestTrackPerturbedVariable(ABC):
+    name: str
+    random_type: RandomType = None
+    lower_bound: float = None
+    upper_bound: float = None
+    historical_forecast_errors: {str: float} = None
+    default: float = None  # [mbar]
```
Comment (zacharyburnettNOAA) [inline ensembleperturbation/perturbation/make_storm_ensemble.py:82]: implement new variables by extending this abstract class (see `CentralPressure`, below, for example)

File: ensembleperturbation/perturbation/make_storm_ensemble.py [hunk 3]
```diff
@@ -0,0 +1,765 @@
...
+
+
+class CentralPressure(BestTrackPerturbedVariable):
+    name = 'central_pressure'
+
+
+class BackgroundPressure(BestTrackPerturbedVariable):
+    name = 'background_pressure'
+    default = 1013.0
```
Comment (zacharyburnettNOAA) [inline ensembleperturbation/perturbation/make_storm_ensemble.py:91]: examples of variable implementations

**Comment by zacharyburnettNOAA** [comment]
@WPringle I've also started using `pint` to handle unit conversions and multiplications between different units.

**Comment by WPringle** [comment]
> @WPringle I've also started using `pint` to handle unit conversions and multiplications between different units.

Yeah that's pretty cool, thanks!

**Comment by WPringle** [comment]
@zacharyburnettNOAA I have a commit where I copied over besttrackforcing from adcircpy into own module in this repository. The test seemed to go through ok (amazingly). 
I was denied when try to push to this branch. What's best to do this? Make PR from a fork of mine to this branch?

**Comment by zacharyburnettNOAA** [comment]
> 
> 
> @zacharyburnettNOAA I have a commit where I copied over besttrackforcing from adcircpy into own module in this repository and removed some adcirc related things. The test seemed to go through ok (amazingly).
> I was denied when try to push to this branch. What's best to do this? Make PR from a fork of mine to this branch?

Yes, for now make a fork + PR, until we can add you to the organization

**Pull request merged**